### DISCUSSION TITLE
"Strict disambiguation" stubs

### DISCUSSION THREAD

**Discussion by ariebovenberg** [ariebovenberg/whenever#280] [Ideas]
With #173 , the `disambiguate` parameter was made optional. For those that want the "stricter" behavior, it might be possible to publish optional stubs that make the parater required. This wouldn't change the actual runtime behavior, but would allow IDEs and typecheckers to enforce the disambiguation parameter.

**Comment by spacemanspiff2007** [comment]
How about an additional option to monkey patch it so it raises an Error like the option to set the time for testing?

```
from whenever import strict_disambiguation

with strict_disambiguation():
    ....
```

**Comment by ariebovenberg** [comment]
Another option is to raise warnings, which can then be ignored by default, but can be turned into errors with the standard warning filter mechanism
